app/models/user.rb

```diff
@@ -9,17 +9,17 @@ class User < ActiveRecord::Base
   attr_accessor(:batch_creation)
   alias :batch_creation? :batch_creation
 
-  has_many :responses, :inverse_of => :user
-  has_many :broadcast_addressings, :inverse_of => :user, :dependent => :destroy
-  has_many :assignments, :autosave => true, :dependent => :destroy, :validate => true, :inverse_of => :user
+  has_many :responses, inverse_of: :user
+  has_many :broadcast_addressings, inverse_of: :user, dependent: :destroy
+  has_many :assignments, autosave: true, dependent: :destroy, validate: true, inverse_of: :user
   has_many :missions, -> { order "missions.created_at DESC" }, through: :assignments
-  has_many :operations, :inverse_of => :creator, :foreign_key => :creator_id, :dependent => :destroy
-  has_many :reports, :inverse_of => :creator, :foreign_key => :creator_id, :dependent => :nullify, class_name: 'Report::Report'
-  has_many :user_groups, :dependent => :destroy
-  has_many :groups, :through => :user_groups
+  has_many :operations, inverse_of: :creator, foreign_key: :creator_id, dependent: :destroy
+  has_many :reports, inverse_of: :creator, foreign_key: :creator_id, dependent: :nullify, class_name: 'Report::Report'
+  has_many :user_group_assignments, dependent: :destroy
+  has_many :user_groups, through: :user_group_assignments, dependent: :destroy
```

Comment:
Have you tested if these delete cleanly on mission delete? Added to mission delete spec? If so, great, disregard.